src-tauri/src/mod_management/mod.rs

```diff
@@ -133,6 +141,7 @@ fn parse_mod_json_for_mod_name(mod_json_path: String) -> Result<String, anyhow::
 }
 
 /// Parses `mod.json` for Thunderstore mod string
+/// This is a legacy function as nowadays `manifest.json` should be in Northtar mods folder and read from there
```

Comment:
Maybe we can even more explicitely set this as deprecated?
https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-deprecated-attribute

Reply:
Hmm, this is more designed for libraries and will throw a compiler warning on every build as long as the function is still in use.

Not super a fan of clogging build with a warning each compile, so as long as the function is still supported, which it is right now, I wouldn't add a deprecation macro to it ^^

Reply:
Hmm I was thinking maybe setting it to a version number greater then current to indicate deprecation in the future but that still causes the warning to show up ^^"

Reply:
Doesn't this show a warning if the deprecated method is used instead?

Reply:
It shows a warning on compile if the deprecated function is used somewhere else in the code. Given that we still need to use it to map Northstar mods to Thunderstore mod that were installed using the old method we will have to call it. So that function technically isn't really deprecated yet. It just exists for legacy compatibility.

The moment the user reinstalls or updates a mod installed using the old method (i.e. writing thunderstore mod string to `mod.json`) it will be overwritten to use the new method (adding `manifest.json` and `thunderstore_author.txt`).

The logic that writes to `mods.json` is not just deprecated but completely removed with this PR as we no longer need it ^^

Reply:
Fine by me :)